<?xml version="1.0"?>

<launch>
  <arg name="joy_pad" default="elecom"/>
  <arg name="DUALSHOCK" default="true"/>
  <arg name="JOY_SMOOTHER" default="false"/>

  <arg name="joy_config" default="elecom-holonomic" />
  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="config_filepath" default="$(find task_programmer)/config/$(arg joy_config).config.yaml" />

  <!-- JoyPad Settings -->
  <node pkg="joy" type="joy_node" name="joy_node" if="$(arg DUALSHOCK)">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.3" />
    <param name="autorepeat_rate" value="20" />
    <remap from="joy" to="$(arg joy_pad)/joy" if="$(arg JOY_SMOOTHER)"/>
  </node>

  <!-- Joy Smoother -->
  <node pkg="task_programmer" type="joy_smoother.py" name="joy_smoother" if="$(arg JOY_SMOOTHER)">
    <remap from="joy_in" to="$(arg joy_pad)/joy" />
    <remap from="joy_out" to="joy" />
  </node>

  <node pkg="teleop_twist_joy" name="teleop_twist_joy" type="teleop_node">
    <rosparam command="load" file="$(arg config_filepath)" />
    <remap from="cmd_vel" to="/gamepad/cmd_vel"/>
  </node>

  <!-- twist mux -->
  <arg name="config_topics" default="$(find task_programmer)/config/cmd_vel.config.yaml"/>
  <node pkg="twist_mux" type="twist_mux" name="twist_mux" output="screen">
    <remap from="cmd_vel_out" to="cmd_vel"/>
    <rosparam file="$(arg config_topics)" command="load"/>
  </node>

</launch>
